extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaCodeGen.java

```diff
@@ -0,0 +1,317 @@
+        final Map<String, String> processedFiles = new HashMap<>();
+        boolean result = false;
+        result |= wsdl2java(context.inputDir(), rootParams, outDir, WSDL2JAVA_CONFIG_KEY_PREFIX, processedFiles);
+
+        final Set<String> names = findParamSetNames(config.getPropertyNames());
+        for (String name : names) {
+            final String prefix = WSDL2JAVA_NAMED_CONFIG_KEY_PREFIX + name;
+            final Wsdl2JavaParameterSet namedParams = buildParameterSet(configFunction,
+                    prefix, null);
+            result |= wsdl2java(context.inputDir(), namedParams, outDir, prefix, processedFiles);
+        }
+
+        return result;
+    }
+
+    static boolean wsdl2java(Path inputDir, Wsdl2JavaParameterSet params, Path outDir, String prefix,
+            Map<String, String> processedFiles) {
+
+        return scan(inputDir, params.includes, params.excludes, prefix, processedFiles, (Path wsdlFile) -> {
+
+            final List<String> args = new ArrayList<>();
+            args.add("-d");
+            args.add(outDir.toString());
+            params.additionalParams.ifPresent(args::addAll);
+            args.add(wsdlFile.toString());
+
+            log.infof("Running wsdl2java %s", args);
+            final ToolContext ctx = new ToolContext();
+            try {
+                new WSDLToJava(args.toArray(new String[0])).run(ctx);
+            } catch (Exception e) {
+                throw new RuntimeException("Could not run wsdl2Java " + args.stream().collect(Collectors.joining(" ")), e);
+            }
+        });
+    }
+
+    static boolean scan(
+            Path inputDir,
+            Optional<List<String>> includes,
+            Optional<List<String>> excludes,
+            String prefix,
+            Map<String, String> processedFiles,
+            Consumer<Path> wsdlFileConsumer) {
+
+        final String selectors = "    " + prefix + ".includes = "
+                + includes.get().stream().collect(Collectors.joining(","))
+                + (excludes.isPresent()
+                        ? "\n    " + prefix + ".excludes = " + excludes.get().stream().collect(Collectors.joining(","))
+                        : "");
+
+        final Consumer<Path> chainedConsumer = wsdlFile -> {
+            final String oldSelectors = processedFiles.get(wsdlFile.toString());
+            if (oldSelectors != null) {
+                throw new IllegalStateException("WSDL file " + wsdlFile + " was already selected by\n\n"
+                        + oldSelectors
+                        + "\n\nand therefore it cannot once again be selected by\n\n" + selectors
+                        + "\n\nPlease make sure that the individual include/exclude sets are mutually exclusive.");
+            }
+            processedFiles.put(wsdlFile.toString(), selectors);
+            wsdlFileConsumer.accept(wsdlFile);
+        };
+
+        final List<Pattern> excludePatterns = excludes.orElse(Collections.emptyList()).stream()
```

Comment:
You may use [DirectoryPathTree](https://github.com/quarkusio/quarkus/blob/main/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/DirectoryPathTree.java) which support multiple include/exclude glob patterns

```
new DirectoryPathTree(inputDir, new PathFilter(includes, excludes)).walk(pathVisit -> {
             Path path = pathVisit.getPath();
               if(Files.isRegularFile(path)) {
                chainedConsumer.accept(visit.getPath());
              }
        });
```

It may be less performant but I don't think users use deep and complex directory hierarchy.

Reply:
Thanks, I did not know, it exists. Is it visible here?